Refactor NBitcoin's ParseIPEndPoint and remove duplicate code

quantumagi opened the pull request:
This PR:
- Refactors NBitcoin's ParseIPEndPoint and removes duplicate code between the method and ToIPEndPoint.

bokobza commented:
some unit tests are failing due to this change:
```
CanParseIpEndpoint
Load_InvalidRpcBindIp_ThrowsConfigurationException
```

quantumagi commented:
Made the following changes:

- Changed `Assert.Throws<FormatException>(() => Utils.ParseIpEndpoint("2001:db8:1f70::999:de8:7648:6e8:100", 90));` from SocketException to FormatException as we would probably not want to try a host name lookup on this value anyway.
- Added support for IP's of the form "1233:3432:2434:2343:3234:2345:6546:4534".

bokobza commented:
Good stuff, thanks mate.
